jsonc/parse.ts

```diff
@@ -51,30 +51,32 @@ export function parse(
   return new JSONCParser(text, { allowTrailingComma }).parse();
 }
 
-enum TokenType {
-  BeginObject,
-  EndObject,
-  BeginArray,
-  EndArray,
-  NameSeparator,
-  ValueSeparator,
-  NullOrTrueOrFalseOrNumber,
-  String,
-}
+const TokenTypeMap = {
+  BeginObject: "BeginObject",
+  EndObject: "EndObject",
+  BeginArray: "BeginArray",
+  EndArray: "EndArray",
+  NameSeparator: "NameSeparator",
+  ValueSeparator: "ValueSeparator",
+  NullOrTrueOrFalseOrNumber: "NullOrTrueOrFalseOrNumber",
+  String: "String",
+} as const;
+
+type TokenType = keyof typeof TokenTypeMap;
```

Comment:
I think it'd be simpler if we just had `TokenType` be a string union type without the intermediary `TokenTypeMap`. A quick skim suggested that might be fine.

Reply:
Done